perf_test/sparse/KokkosSparse_spiluk.cpp

```diff
@@ -0,0 +1,500 @@
+                              subview(h_L_values,    Kokkos::make_pair(l_row_start,l_row_end)));//L part
+            Kokkos::deep_copy(subview(h_tmp_values,  Kokkos::make_pair(l_row_end-l_row_start-1,a_row_end-a_row_start)),
+                              subview(h_U_values,    Kokkos::make_pair(u_row_start,u_row_end+1)));//U part
+        
+            for ( size_type k = 0; k < (a_row_end-a_row_start); ++k ) {
+              if ( h_tmp_entries(k) != h_A_entries(a_row_start+k) ) {
+                if ( h_A_entries(a_row_start+k) < i )
+                  std::cout << "ILU(0) FAILURE: non-zero col idx on row " << i << " do not match -- KK (L part) = " << h_tmp_entries(k) << ", cuSPARSE = " << h_A_entries(a_row_start+k) << std::endl;
+                else
+                  std::cout << "ILU(0) FAILURE: non-zero col idx on row " << i << " do not match -- KK (U part) = " << h_tmp_entries(k) << ", cuSPARSE = " << h_A_entries(a_row_start+k) << std::endl;
+                return 1;
+              } else if ( abs(h_tmp_values(k) - h_A_values(a_row_start+k)) > 1e-9 ) {
+                if ( h_A_entries(a_row_start+k) < i )
+                  std::cout << "ILU(0) FAILURE: non-zero entry on row " << i << "do not match -- KK (L part) = " << h_tmp_values(k) << " at col " << h_tmp_entries(k) << ", cuSPARSE = " << h_A_values(a_row_start+k) << " at col " << h_A_entries(a_row_start+k) << std::endl;
+                else
+                  std::cout << "ILU(0) FAILURE: non-zero entry on row " << i << "do not match -- KK (U part) = " << h_tmp_values(k) << " at col " << h_tmp_entries(k) << ", cuSPARSE = " << h_A_values(a_row_start+k) << " at col " << h_A_entries(a_row_start+k) << std::endl;
+                return 1;
+              }
+            }// end col
+          }
+        }// end row
+        std::cout << "ILU(0) SUCCESS!" << std::endl;
+     }//fill_lev=0
+#endif
+
+      // Benchmark
+      Kokkos::fence();
+      double min_time = 1.0e32;
+      double max_time = 0.0;
+      double ave_time = 0.0;
+      
+      for(int i=0;i<loop;i++) {
+        timer.reset();
+        spiluk_numeric( &kh, fill_lev, 
+                        A.graph.row_map, A.graph.entries, A.values, 
+                        L_row_map, L_entries, L_values, U_row_map, U_entries, U_values );
+        Kokkos::fence();
+        double time = timer.seconds();
+        ave_time += time;
+        if(time>max_time) max_time = time;
+        if(time<min_time) min_time = time;
+      }
+      std::cout << "LOOP_AVG_TIME:  " << ave_time/loop << std::endl;
+      std::cout << "LOOP_MAX_TIME:  " << max_time << std::endl;
+      std::cout << "LOOP_MIN_TIME:  " << min_time << std::endl;
+
+#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+      if (fill_lev==0) {
+        
+        lno_view_t     A_row_map("A_row_map", nrows + 1);
+        lno_nnz_view_t A_entries("A_entries", nnz);
+        scalar_view_t  A_values ("A_values",  nnz);
+
+        min_time = 1.0e32;
```

Comment:
You've included `<limits>`, so you should use `std::numeric_limits<double>::infinity()` here instead of this arbitrary "big number."

Reply:
@mhoemmen. Thanks. It is fixed.